### DISCUSSION TITLE
create 32bit wine prefix on game install

### DISCUSSION THREAD

**Discussion by toratora** [Heroic-Games-Launcher/HeroicGamesLauncher#2752] [Q&A]
I'm attempting to install "Axiom Verge 2" from EGS via Heroic 2.8.0. According to a ProtonDB entry on the game :

"If you only have this through Epic and not Steam, the native Linux versions isn't available. The Windows version will NOT run through Proton. You can use Heroic Launcher to install the non-proton version of WINE GE and add WINEARCH=win32 in the environment variables under "Other options" before running it to create a 32 bit wine prefix, then use WINETRICKS to install dotnet452."

Is there a way to set the WINEARCH environment variable before the wine prefix is created so that a 32 bit wine prefix is created? I attempted to set the variable afterwards but winetricks still says it's a 64bit prefix and displays a bunch of warnings while installing dotnet452. 

The game failed to launch and the log indicated that WINEARCH=win32 was set but that the prefix was 64 bit.

BTW, I was using Wine version "Wine - Wine - 7.14-GO-1-LoL" which seemed like one of the few "non-proton" wine versions available for use.

Any help would be appreciated.

**Comment by sudence** [comment]
no solutions yet :-:

**Reply by mvanthoor** [reply]
You can't define a 32-bit Wine prefix on game install, but there is a workaround.

- First, make sure you have a Wine/Proton version installed that supports 32-bit (I use Wine-GE-Proton_7.43 from the Wine Manager, because I have a game that needs a Wine-version older than 8)
- Second, in Settings -> Game Defaults -> Advanced, set this variable: WINEARCH = win32
- Then install your game like normal. Make sure you select the Wine version that supports 32-bit.

Every prefix you create will be 32-bit however, so you have to remove the variable WINEARCH after you're done.

I installed a Wine 32-bit prefix like this for Baldur's Gate and Icewind Dale, porting them over from Lutris.  Both games need a 32-bit prefix with Wine 7.x (or older), with Windows 98 set to function properly. They will work in a 64-bit Proton prefix, but then they have massive mouse flickering and scrolling issues which I can't get solved; and I want to finish these runs because they're both already at 90%.

**Comment by Proficiency-git** [comment]
Thanks for the tip. 

If you can't fix the mouse flickering, try to enable the wine-wayland option on the launcher setting for the game. I use GNOME with Wayland so I think that is also a factor as to why that option fix it.

But I do have a problem I can't solve: My Keyboard doesn't work! (I'm playing Icewind Dale 2)

Being able to play my Infinity Engine games (non-enhanced) is the last barrier for me in term of Linux Gaming. I'm going to try your tip and reinstall the game but I doubt I could fix this specific problem, weird thing is, keyboard works on BG2 even using proton-GE (10x), I just had the mouse flickering problem which I can't solve when I had it installed. 

Edit, I reinstalled the game with your tip and it works like a charm, Baldur's Gate 2 is running like a dream, like it was supposed to be in holy 4:3 800x600 resolution, but the keyboard still doesn't work for IWD2.
